jtuples/src/test/java/org/jtuples/FunctionalTest.java

```diff
@@ -0,0 +1,139 @@
+        BiFunction<String, String, String> f =
+                Functional.curry(p -> p.first() + p.second());
+        assertEquals(f.apply("hello ", "world"), "hello world");
+    }
+
+    /**
+     * Test of uncurry method, of class Functional.
+     */
+    @Test
+    public void testUncurry() {
+        Function<Pair<String, String>, String> f =
+                Functional.uncurry((a, b) -> a + b);
+        assertEquals(f.apply(new Pair<>("hello ", "world")), "hello world");
+    }
+
+    /**
+     * Test of split method, of class Functional.
+     */
+    @Test
+    public void testSplit_Function_Function() {
+        Function<String, Pair<String, Integer>> f = 
+                Functional.split(s -> s.trim(), s -> s.length());
+        assertEquals(f.apply("  s  "), new Pair<>("s", 5));
+    }
+
+    /**
+     * Test of split method, of class Functional.
+     */
+    @Test
+    public void testSplit_BiFunction_BiFunction() {
+        BiFunction<Integer, Integer, Pair<Integer, Integer>> f =
+                Functional.split((a, b) -> a + b, (a, b) -> a * b);
+        assertEquals(f.apply(2, 3), new Pair<>(5, 6));
+    }
+
+    /**
+     * Test of split method, of class Functional.
+     */
+    @Test
+    public void testSplit_Supplier_Supplier() {
+        Supplier<Pair<Integer, Integer>> f =
+                Functional.split(() -> 1, () -> 2);
+        assertEquals(f.get(), new Pair<>(1, 2));
+    }
+
+    /**
+     * Test of product method, of class Functional.
+     */
+    @Test
+    public void testProduct_Function_Function() {
+        Function<Pair<Integer, Integer>, Pair<Integer, Integer>> f =
+                Functional.product(a -> a + 1, b -> 2 * b);
+        assertEquals(f.apply(new Pair<>(1, 2)), new Pair<>(2, 4));
+    }
+
+    /**
+     * Test of product method, of class Functional.
+     */
+    @Test
+    public void testProduct_BiFunction_BiFunction() {
+        BiFunction<Pair<Integer, Integer>, Pair<Integer, Integer>,
+                Pair<Integer, Integer>> f =
+                Functional.product((a, b) -> a + b, (a, b) -> a * b);
+        assertEquals(f.apply(new Pair<>(1, 2), new Pair<>(3, 4)),
+                new Pair<>(4, 8));
+    }
+
+    /**
+     * Test of product method, of class Functional.
+     */
+    @Test
+    public void testProduct_Supplier_Supplier() {
+        Supplier<Pair<Integer, Integer>> f =
+                Functional.product(() -> 1, () -> 2);
+        assertEquals(f.get(), new Pair<>(1, 2));
+    }
+
+    /**
+     * Test of product method, of class Functional.
+     */
+    @Test
+    public void testProduct_Consumer_Consumer() {
+        Consumer<Pair<Integer, Integer>> f = Functional.product(
+                a -> {
+                    if (!a.equals(1)) throw new IllegalArgumentException();
+                },
+                b -> {
+                    if (!b.equals(2)) throw new IllegalArgumentException();
+                });
+        f.accept(new Pair<>(1, 2));
```

Comment:
This test doesn't have any assertion or expectation (which means it's not actually a test).
Maybe you forgot to [expect an exception](https://github.com/junit-team/junit/wiki/Exception-testing#expectedexception-rule)?